src/KafkaNET.Library/Consumers/PartitionTopicInfo.cs

```diff
@@ -218,6 +220,39 @@ public long FetchOffset
         // Gets or sets the offset for the next request.
         public long NextRequestOffset { get; internal set; }
 
+        public bool ConsumedOffsetOutOfRange
+        {
+            get
+            {
+                lock(consumedOffsetOutOfRangeLock)
+                {
+                    return this.consumedOffsetOutOfRangeFlag;
+                }
+            }
+            internal set
+            {
+                lock (consumedOffsetOutOfRangeLock)
+                {
+                    this.consumedOffsetOutOfRangeFlag = value;
+
+                    if (this.consumedOffsetOutOfRangeFlag)
+                    {
+                        // Clear the chunkQueue as current messages in the queue are invalid on brokers. Drop them in order to fetch valid messages in.
+                        while (this.chunkQueue.Count > 0)
```

Comment:
Dequeuing actually guarantee that the size of the queue are no bigger than the number we actually set. If we don't dequeue, the worst case is when it keeps getting out of range error(which will happen as out of range is due to consumer cannot catch up), the size of the queue will keep growing till it blows up.